app/controllers/posts_controller.rb

```diff
@@ -1,31 +1,61 @@
 class PostsController < ApplicationController
+  load_and_authorize_resource
+
   def index
     @posts = Post.all
     @user = User.includes(posts: [:comments]).find(params[:user_id].to_i)
   end
 
+  def current_ability
+    @current_ability ||= Ability.new(current_admin)
+  end
+
   def new
+    @id = current_admin.id
     @post = Post.new
+    @id = current_admin.id
   end
 
   def create
-    @post = Post.new(post_params)
-    @author = current_user
-    @post.author = @author
-    if @post.save
-      flash[:success] = 'Post was successfully created'
-      redirect_to new_user_post_path(@author.id)
+    post = Post.new(title: post_params[:title], text: post_params[:text], author: current_admin)
+    @author = post.author
+
+    if post.save
+      redirect_to new_user_post_path(@author.id), message: 'Post was successfully created'
     else
       flash[:error] = 'Error:  Post was not created!!'
       render :new
     end
   end
 
+  # def create
+  #   @post = Post.new(post_params)
+  #   @author = current_user
+  #   @post.author = @author
+  #   if @post.save
+  #     flash[:success] = 'Post was successfully created'
+  #     redirect_to new_user_post_path(@author.id)
+  #   else
+  #     flash[:error] = 'Error:  Post was not created!!'
+  #     render :new
+  #   end
+  # end
+
```

Comment:
Kindly remove commented code that you do not need with respect to the YAGNI principle. This will keep your code clean and readable.

Reply:
commented code removed.